### PR TITLE
bugfix the callback behaviors.

### PR DESCRIPTION
**Pull request by alfa-jpn**
bugfix the callback behaviors. #81

and add tests for plain models callback.

### CODE REVIEW

File: lib/paranoia.rb [hunk 1]
```diff
@@ -67,11 +67,20 @@ def destroyed?
...
+  # insert time to paranoia column.
+  # @param with_transaction [Boolean] exec with ActiveRecord Transactions.
+  def touch_paranoia_column(with_transaction=false)
+    return touch(paranoia_column) unless with_transaction
 
-  # Rails 3.1 adds update_column. Rails > 3.2.6 deprecates update_attribute, gone in Rails 4.
-  def update_attribute_or_column(*args)
-    self.class.unscoped do
-      respond_to?(:update_column) ? update_column(*args) : update_attribute(*args)
+    with_transaction_returning_status do
```
Comment (radar) [inline lib/paranoia.rb]: This looks to me like it would call `touch` twice.

Comment (alfa-jpn) [inline lib/paranoia.rb]: How about this?

``` ruby
def touch_paranoia_column(with_transaction=false)
  if with_transaction
    with_transaction_returning_status { touch(paranoia_column) }
  else
    touch(paranoia_column)
  end
end
```

or

``` ruby
def touch_paranoia_column(with_transaction=false)
  return with_transaction_returning_status { touch_paranoia_column(false) } if with_transaction
  touch(paranoia_column)
end
```

Comment (radar) [inline lib/paranoia.rb]: The first example looks the cleanest to me. Could you please implement that?

The second example has exactly the same bug of calling `touch` twice, so no that is not a good solution.

**Comment by alfa-jpn** [comment]
I changed touch_paranoa_column to the discussed code.

**Comment by radar** [comment]
Well, the CI server seems to be happy with this and the code looks fine to me. Thanks!

**Comment by radar** [comment]
Added to master and rails4.

**Pull request closed without merging**